indico/filters/base.py

```diff
@@ -0,0 +1,34 @@
+from enum import Enum
+from typing import Any, Dict, Iterable
+
+
+class BaseFilter(dict):
+    """
+    Base filter class that allows users to construct filter statements for
+    GraphQL queries. Columns are restricted to a particular set of database columns
+    determined by the specific resource a user queries.
+
+    Note:
+        Class should be extended for specific Indico resources, and the subclasses 
+        instantiated when a query is constructed.
+    """
+
+    _filterable_columns = set()
+
+    def __init__(self, filters: Dict[str, Any]):
+        self._validate(filters)
+        super().__init__(**filters)
+
+    @property
+    def filterable_columns(self):
+        return self._filterable_columns
+
+    @classmethod
+    def _validate(cls, keys: Iterable):
+        if not all(key in cls._filterable_columns for key in keys):
```

Comment:
hmm, filters can take `AND` and `OR` as well
For example: https://github.com/IndicoDataSolutions/sunbow/blob/REPAIR/tests/routes/test_submission.py#L272
here I filter for 2 different statuses

Reply:
this is just checking that the user is actually filtering on the correct columns, as defined by each particular resource.

Reply:
I see what you mean ...